foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/mappings/UnidirectionalOneToManyMapping.java

```diff
@@ -390,4 +392,42 @@ public boolean shouldIncrementTargetLockValueOnDeleteSource() {
     protected boolean shouldRemoveTargetQueryModifyTargetForeignKey() {
         return true;
     }
+
+    /**
+     * INTERNAL: Merges the given event with the given query. This means that the query will contain the applicable event rows.
+     * 
+     * @param query
+     * @param event
+     * @param session
+     * @throws DatabaseException
+     * @throws DescriptorException
+     */
+    public boolean mergeDataModificationEvent(WriteObjectQuery query, Object[] event, AbstractSession session) throws DatabaseException, DescriptorException {
+        if (event[0] == PostInsert) {
+        } else if (event[0] == ObjectRemoved) {
+        } else if (event[0] == ObjectAdded) {
+            WriteObjectQuery wq = (WriteObjectQuery)event[1];
+
+            AbstractRecord queryModifyRow = query.getModifyRow();
+            AbstractRecord updateTransRow = buildKeyRowForTargetUpdate(wq);
+
+            // Look for any rows that are non-nullable, we need to add them to the query
+            for(int pos = 0; pos < updateTransRow.getFields().size(); pos++) {
+                DatabaseField fkField = updateTransRow.getFields().get(pos);
+                if(fkField != null && !fkField.isNullable()) {
```

Comment:
Dusting off the old EclipseLink knowledge but is this flag set when the database is pre-existing?  I don't remember a feature in core that read in the schemas.  The code should just include all FK fields where this mapping is the owner.

Reply:
This flag is set during orm processing. We look at the @JoinColumn annotation and set the flag based on the value in the annotation! If the database is pre-existing, I think it should be the responsibility of the user to have their orm configuration match their database. 

Essentially, during UPDATE for POST (owner of the UnidirectionalOneToMany mapping), we create a future event also UPDATE Comment (specifically the FKs). My change is intended to merge any non-nullable FKs into the INSERT for Comment. I'm doing it this way (instead of using all FKs) because we can just let the UPDATE for Comment occur as it already works. This change should only alter behavior for 'nullable=false' mappings.